src/base/emu-i386/kvm.c

```diff
@@ -1021,6 +1045,41 @@ static unsigned int kvm_run(void)
       break;
     case KVM_EXIT_MMIO:
       /* for ROM: simply ignore the write and continue */
+      if (memcheck_is_rom(run->mmio.phys_addr))
+	break;
+
+      /* from the KVM api.txt: "the corresponding operations are complete
+	 (and guest state is consistent) only after userspace has re-entered
+	 the kernel with KVM_RUN. The kernel side will first finish
+	 incomplete operations and then check for pending signals." */
+      run->immediate_exit = 1;
+      do {
+	dosaddr_t addr = (dosaddr_t)run->mmio.phys_addr;
+	unsigned char *data = run->mmio.data;
+	if (run->mmio.is_write) {
+	  switch(run->mmio.len) {
+	  case 1: write_byte(addr, data[0]); break;
+	  case 2: write_word(addr, *(uint16_t*)data); break;
+	  case 4: write_dword(addr, *(uint32_t*)data); break;
```

Comment:
kvm seems to allow 8 too.
I suspect we can't hit that, but at least
a "default" catch would be good.
Can some rep movs give 8 btw?